WebDriverAgentLib/Utilities/FBXPath.m

```diff
@@ -309,7 +309,7 @@ + (int)writeXmlWithRootElement:(id<FBElement>)root indexPath:(nullable NSString
       NSArray<XCUIElement *> *windows = [((XCUIElement *)root) fb_filterDescendantsWithSnapshots:currentSnapshot.children];
       NSMutableArray<XCElementSnapshot *> *windowsSnapshots = [NSMutableArray array];
       for (XCUIElement* window in windows) {
-        [windowsSnapshots addObject:window.fb_lastSnapshot];
+        [windowsSnapshots addObject:window.fb_snapshotWithAttributes ?: window.fb_lastSnapshot];
       }
       children = windowsSnapshots.copy;
     } else {
```

Comment:
Line 316 shouldn't be reached when getting the page source because we will always have an application as the root.
And I'm not sure about line 308 since the children are requested separately. Requesting all attributes for the root and it's children, but not using them looks like a bit of overhead.

Reply:
this must be in sync, since the stuff is also used for xpath search. which means the lookup might behave differently based on whether you execute it on application or in element context.